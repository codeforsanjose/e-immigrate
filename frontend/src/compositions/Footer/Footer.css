.footer {
    background-color: var(--color-brightSky);
    padding: 18px 2rem;
    display: grid;
    grid-template-rows: 50% 35% 15%;
    grid-template-rows: auto;
    align-items: center;
    justify-content: center;
    overflow: none;
}

.footer .imgRow {
    display: grid;
    grid-template-columns: auto auto;
    gap: 18px;
    align-items: center;
}

.footer .imgRow .ciitLogo {
    width: 48px;
    height: auto;
    justify-self: end;
}

.footer .imgRow .ceticpLogo {
    width: 48px;
    height: auto;
    justify-self: start;
}

.footer .textRow1 {
    text-align: center;
    max-width: 500px;
    margin-bottom: 12px;
}

.footer .textRow2 {
    text-align: center;
}
