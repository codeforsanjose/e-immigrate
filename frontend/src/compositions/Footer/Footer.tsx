import React from 'react';
import ciitLogo from '../../data/images/CiiT Logo.png';
import ceticpLogo from '../../data/images/CET-ICP Logo.png';
import './Footer.css';

type FooterProps = {
    content?: {
        footerText1?: string;
        footerText2?: string;
    };
};
export function Footer(props: FooterProps) {
    const { content = { footerText1: '', footerText2: '' } } = props;
    const { footerText1, footerText2 } = content;
    return (
        <div className="footer">
            <div className="imgRow">
                <img className="ciitLogo" src={ciitLogo} alt="CiiT Logo" />
                <img
                    className="ceticpLogo"
                    src={ceticpLogo}
                    alt="CET-ICP Logo" />
            </div>
            <div className="textRow1">{footerText1}</div>
            <div className="textRow2">{footerText2}</div>
        </div>
    );
}
