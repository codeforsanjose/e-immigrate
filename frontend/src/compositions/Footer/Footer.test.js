import React from 'react';
import Footer from './Footer';

import renderer from 'react-test-renderer';

const content = {
    footerText1: 'test',
    footerText2: 'test',
};

it('Footer renders correctly', () => {
    const tree = renderer.create(<Footer content={content} />).toJSON();
    expect(tree).toMatchSnapshot();
});
