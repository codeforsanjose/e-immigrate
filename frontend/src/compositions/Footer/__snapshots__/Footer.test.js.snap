// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="footer"
>
  <div
    className="imgRow"
  >
    <img
      alt="CiiT Logo"
      className="ciitLogo"
      src="CiiT Logo.png"
    />
    <img
      alt="CET-ICP Logo"
      className="ceticpLogo"
      src="CET-ICP Logo.png"
    />
  </div>
  <div
    className="textRow1"
  >
    test
  </div>
  <div
    className="textRow2"
  >
    test
  </div>
</div>
`;
